control-plane-operator/controllers/awsprivatelink/awsprivatelink_controller.go

```diff
@@ -117,9 +120,14 @@ func (r *PrivateServiceObserver) Reconcile(ctx context.Context, req ctrl.Request
 	if err := r.List(ctx, hcpList, &client.ListOptions{Namespace: r.HCPNamespace}); err != nil {
 		return ctrl.Result{}, fmt.Errorf("failed to get resource: %w", err)
 	}
-	if len(hcpList.Items) != 1 {
+	if len(hcpList.Items) == 0 {
+		// Return early if HostedControlPlane is deleted
+		return ctrl.Result{}, nil
+	}
+	if len(hcpList.Items) > 1 {
 		return ctrl.Result{}, fmt.Errorf("unexpected number of HostedControlPlanes in namespace, expected: 1, actual: %d", len(hcpList.Items))
```

Comment:
Just an observation (this code was already there so not sure we need to change anything yet here)... if this condition arises, seems like reconcilation will simply silently halt forever. Surfacing some kind of event or condition that the user can actually find might be good...

Reply:
This is really just for completeness.  I think if there were more than one HCP in a namespace, the world would go boom in many ways.